### PR TITLE
chore(minor): Reduce dependencies

### PR DESCRIPTION
**Pull request by hassila**
## Description

Reduce dependencies from 12->7;

Removing extras-json and instead use Foundations JSON - we have a Foundation dependency anyway and will hopefully soon move to the new swift-foundation.

Fold in Progress.Swift inline with the project instead of depending on Ordo tagged release version 

Fold in BenchmarkClock to remove dependency on Ordo package-datetime

Update NIO locks from upstream

Move Benchmarks to a separate embedded project a la SwiftNIO to allow it to use newer toolchain and remove dependencies

Remove DocC plugin dependency as it is done behind the scenes by SPI anyway

## How Has This Been Tested?

Test suite and manually tested on macOS 

## Minimal checklist:

- [x] I have performed a self-review of my own code 
- [ ] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Sources/Benchmark/NIOConcurrencyHelpers/lock.swift [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+#endif
+
+/// A threading lock based on `libpthread` instead of `libdispatch`.
+///
+/// This object provides a lock on top of a single `pthread_mutex_t`. This kind
+/// of lock is safe to use with `libpthread`-based threading models, such as the
+/// one used by NIO. On Windows, the lock is based on the substantially similar
+/// `SRWLOCK` type.
+@available(*, deprecated, renamed: "NIOLock")
+public final class Lock {
```
Comment (dimlio) [inline Sources/Benchmark/NIOConcurrencyHelpers/lock.swift:35]: This implementation is deprecated and will be removed later, right?

**Review by dimlio — APPROVED** [review]

File: Sources/Benchmark/NIOConcurrencyHelpers/lock.swift (hunk 1 above)
Comment (hassila) [reply Sources/Benchmark/NIOConcurrencyHelpers/lock.swift:35]: We just lift the implementation as-is from NIO right now until we get lock support from the standard library, so those two files are just copy-pasted basically and updated from time-to-time.

File: Sources/Benchmark/BenchmarkClock.swift [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+#if os(macOS) || os(iOS)
+        let nanos = clock_gettime_nsec_np(CLOCK_UPTIME_RAW) // to get ns resolution on macOS
+
+        let seconds: UInt64 = nanos / 1_000_000_000
+        let attoseconds: UInt64 = (nanos % 1_000_000_000) * 1_000_000_000
+        return BenchmarkClock.Instant(_value: Duration(secondsComponent: Int64(seconds),
+                                                       attosecondsComponent: Int64(attoseconds)))
+#elseif os(Linux)
+        var timespec = timespec()
+        let result = clock_gettime(CLOCK_REALTIME, &timespec)
```
[the remark this answers is not in the document]
Comment (hassila) [reply Sources/Benchmark/BenchmarkClock.swift:86]: Yeah, thanks.

File: Package.swift [hunk 3]
```diff
@@ -18,19 +18,15 @@ let package = Package(
...
-//        .package(url: "https://github.com/SwiftPackageIndex/SPIManifest", from: "0.12.0"),
         .package(url: "https://github.com/ordo-one/TextTable", .upToNextMajor(from: "0.0.1")),
-        .package(url: "https://github.com/ordo-one/package-datetime", .upToNextMajor(from: "1.0.1")),
-        .package(url: "https://github.com/ordo-one/package-histogram", .upToNextMajor(from: "0.0.1")),
-        .package(url: "https://github.com/ordo-one/Progress.swift", .upToNextMajor(from: "1.0.0")),
-        .package(url: "https://github.com/apple/swift-docc-plugin", .upToNextMajor(from: "1.1.0")),
-        .package(url: "https://github.com/apple/swift-atomics", .upToNextMajor(from: "1.0.0")),
+        .package(url: "https://github.com/HdrHistogram/hdrhistogram-swift", .upToNextMajor(from: "0.1.0")),
+        .package(url: "https://github.com/apple/swift-atomics", .upToNextMajor(from: "1.0.0"))
+        //        .package(url: "https://github.com/SwiftPackageIndex/SPIManifest", from: "0.12.0"),
```
[the remark this answers is not in the document]
Comment (hassila) [reply Package.swift:29]: I removed it for now.

File: Sources/Benchmark/BenchmarkClock.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (hassila) [reply Sources/Benchmark/BenchmarkClock.swift]: Actually @dimlio, I moved to `CLOCK_BOOTTIME` instead, which also takes suspends into account.

**Pull request merged**